common/tools/dev-tool/src/util/fsHelpers.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { access, constants } from "node:fs/promises";
+
+/**
+ * Check if a file or directory exists at the given path.
+ * This is a replacement for fs-extra's pathExists method.
+ *
+ * @param path - The path to check
+ * @returns A promise that resolves to true if the path exists, false otherwise
+ */
+export async function pathExists(path: string): Promise<boolean> {
+  try {
+    await access(path, constants.F_OK);
```

Comment:
two things that are interesting from https://nodejs.org/docs/latest/api/fs.html#fspromisesaccesspath-mode

1. F_OK is the default mode, so no real reason to pass it explicitly
2. The docs say not to use it to guard against a file not being available for read:

> Using fsPromises.access() to check for the accessibility of a file before calling fsPromises.open() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

Reply:
For our dev-tool and perf test utils, the sync version `existsSync` should be fine.